### DISCUSSION TITLE
Will Homebrew uses tools from Xcode CLT if needed although Xcode.app is set as active development directory?

### DISCUSSION THREAD

**Discussion by leungyauming** [Homebrew/discussions#2402] [Everyday usage] [answered]
I ask if I will be able to use Xcode.app only for Homebrew in #2335 .
But I have another question, will Homebrew uses tools from Xcode CLT if needed although Xcode.app is set as active development directory?

**Comment by Bo98** [comment, marked as answer]
The SDKs from the CLT will always be prioritised. For the compiler etc, we just use whatever you have set as the active directory - it doesn't matter much if both are up-to-date.

We _require_ the CLT on macOS versions that are not the latest because Xcode does not ship SDKs for older macOS versions, even if Xcode itself supports those versions. It is however also required on even the latest macOS for some formulae like LLVM and Python (unless you build them from source yourself) because they are built with baked in paths pointing to the SDK and that can't be Xcode because Xcode.app isn't guaranteed to be at a fixed path (but the CLT is).

Our SDK selection completely bypasses your developer directory setting, so you don't need to worry about what your developer directory is set to. That setting will mostly affect the compiler rather than anything SDK related. In fact, our CI machines have Xcode set as the developer directory.

**Reply by leungyauming** [reply]
Ok understand, thank you for the quick reply.
